packages/react/src/hooks/utils/query/useMutation.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  MutationFunction,
+  MutationKey,
+  UseMutationOptions,
+  UseMutationResult,
+  parseMutationArgs,
+  useMutation as rqUseMutation,
+} from '@tanstack/react-query'
+
+import { queryClientContext as context } from '../../../context'
+
+export function useMutation<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
+  TContext = unknown,
+>(
+  options: UseMutationOptions<TData, TError, TVariables, TContext>,
+): UseMutationResult<TData, TError, TVariables, TContext>
+
+export function useMutation<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
+  TContext = unknown,
+>(
+  mutationFn: MutationFunction<TData, TVariables>,
+  options?: Omit<
+    UseMutationOptions<TData, TError, TVariables, TContext>,
+    'mutationFn'
+  >,
+): UseMutationResult<TData, TError, TVariables, TContext>
+
+export function useMutation<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
+  TContext = unknown,
+>(
+  mutationKey: MutationKey,
+  options?: Omit<
+    UseMutationOptions<TData, TError, TVariables, TContext>,
+    'mutationKey'
+  >,
+): UseMutationResult<TData, TError, TVariables, TContext>
+
+export function useMutation<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
+  TContext = unknown,
+>(
+  mutationKey: MutationKey,
+  mutationFn?: MutationFunction<TData, TVariables>,
+  options?: Omit<
+    UseMutationOptions<TData, TError, TVariables, TContext>,
+    'mutationKey' | 'mutationFn'
+  >,
+): UseMutationResult<TData, TError, TVariables, TContext>
+
+export function useMutation<
+  TData = unknown,
+  TError = unknown,
+  TVariables = void,
+  TContext = unknown,
+>(
+  arg1:
+    | MutationKey
+    | MutationFunction<TData, TVariables>
+    | UseMutationOptions<TData, TError, TVariables, TContext>,
+  arg2?:
+    | MutationFunction<TData, TVariables>
+    | UseMutationOptions<TData, TError, TVariables, TContext>,
+  arg3?: UseMutationOptions<TData, TError, TVariables, TContext>,
+): UseMutationResult<TData, TError, TVariables, TContext> {
+  const options = parseMutationArgs(arg1, arg2, arg3)
+  return rqUseMutation({ context, ...options })
+}
```

Comment:
We should export `useMutation` from `wagmi` too (just like `useQuery`).

Reply:
I'm not sure I understand this one.  Can you provide some more information on this?

Reply:
```ts
// packages/react/src/index.ts

export {
  ...
  useMutation,
  ...
} from './hooks'
```

Reply:
Ahh! yes.  Makes sense.

Reply:
`useQueryClient` also?  If the rational is for an "escape hatch" then I think that makes sense.